develop/reference_workarounds.rst

```diff
@@ -67,6 +67,34 @@ Although you can't join such tables directly, by wrapping the local table in a s
 
 Remember that the coordinator will send the results in the subquery or CTE to all workers which require it for processing. Thus it's best to either add the most specific filters and limits to the inner query as possible, or else aggregate the table. That reduces the network overhead which such a query can cause. More about this in :ref:`subquery_perf`.
 
+.. _upsert_into_select:
+
+INSERT…SELECT upserts
+---------------------
+
+For those INSERT…SELECT statements including a GROUP BY clause which contains the distribution column, Citus can push the whole statement down to workers for execution. Otherwise Citus uses :ref:`push_pull_execution` to pull the results of the SELECT query from workers to the coordinator, and push the rows for insertion back down to the workers. This latter mode does not support the ON CONFLICT clause (Postgres' way to do upserts).
```

Comment:
A correction here. We need to make sure we select partition column from one table into the partition column of another table. If there is an agg, then the partition column needs to be in the group-by clause.
See https://docs.google.com/presentation/d/1mxoLq5d8-jhO9XjYR7I4oJCUo5r2eaqSZPNkepO4yQc/edit#slide=id.g18d75ba220_0_28. Could we amend this description?